File: frontend/src/types/team.ts
/**
 * Team and invitation types for the VibeXP frontend
 */

/**
 * Status of a team invitation
 */
export type InvitationStatus = 'pending' | 'accepted' | 'rejected' | 'expired'

/**
 * Team member role
 */
export type TeamRole = 'owner' | 'admin' | 'member'

/**
 * Team member details
 */
export interface TeamMember {
  user_id: string
  email: string
  name: string
  role: TeamRole
  joined_at: string
  invitation_status?: 'pending' | 'accepted'
}

/**
 * Team details
 */
export interface Team {
  id: string
  name: string
  slug: string
  description: string
  // role may be absent from API responses in some contexts (e.g. when fetching
  // team details without membership context). Components must guard against
  // undefined (VIBEXP-FRONTEND-JS-5).
  role?: TeamRole
  member_count: number
  is_personal: boolean
  created_at: string
  updated_at: string
}

/**
 * Team with full member list
 */
export interface TeamWithMembers extends Team {
  members: TeamMember[]
}

/**
 * Team invitation details.
 *
 * The backend wire format uses `invitee_email` (see `models.InvitationResponse`).
 * `email` is a legacy alias kept for older consumers; new code should read
 * `invitee_email`. Both are optional on the type because different endpoints
 * historically populated different fields — readers must guard against either
 * being undefined.
 */
export interface TeamInvitation {
  id: string
  token: string
  team_id: string
  team_name: string
  /** Backend wire format field (canonical). */
  invitee_email?: string
  /** Legacy alias kept for existing consumers. */
  email?: string
  role?: TeamRole
  status: InvitationStatus
  created_at: string
  expires_at: string
  invited_by?: {
    id: string
    name: string
    email: string
  }
}

/**
 * Request to create a new team
 */
export interface CreateTeamRequest {
  name: string
  description?: string
}

/**
 * Request to update a team
 */
export interface UpdateTeamRequest {
  name?: string
  description?: string
}

/**
 * Request to invite team members
 */
export interface InviteTeamMembersRequest {
  emails: string[]
  role?: 'member' | 'admin'
}

/**
 * Response for team list
 */
export interface TeamsResponse {
  teams: Team[]
  total_count: number
  page: number
  page_size: number
}

/**
 * Response for team members
 */
export interface TeamMembersResponse {
  // members may be absent from the API response when the team has no members
  // or when the field is omitted by the server (VIBEXP-FRONTEND-JS-6).
  members?: TeamMember[]
  total_count: number
  page: number
  page_size: number
}

/**
 * Response for invite members
 */
export interface InviteTeamMembersResponse {
  invitations: {
    email: string
    invitation_id: string
    status: string
  }[]
  success_count: number
  error_count: number
}

/**
 * Response for a single invitation
 */
export interface InvitationResponse {
  invitation: TeamInvitation
}

/**
 * Response for pending invitations list
 */
export interface PendingInvitationsResponse {
  // invitations may be absent from the API response when the server omits the
  // field for users with no invitations (VIBEXP-FRONTEND-JS-4).
  invitations?: TeamInvitation[]
  total_count: number
  page: number
  page_size: number
}

/**
 * Accept invitation response
 */
export interface AcceptInvitationResponse {
  team_id: string
  team_name: string
  message: string
}

/**
 * Team-wide resource counts for the team analytics page. Mirrors the backend
 * TeamStatsResponse — like ProjectStats but team-scoped and with a projects count.
 */
export interface TeamStats {
  total_projects: number
  total_prompts: number
  total_artifacts: number
  total_blueprints: number
  total_memories: number
  total_feed_items: number
}

File: backend/internal/models/team.go
package models

import "time"

// Team represents a team in the system
type Team struct {
	ID          string    `json:"id" db:"id"`
	OwnerID     string    `json:"owner_id" db:"owner_id"`
	Name        string    `json:"name" db:"name"`
	Slug        string    `json:"slug" db:"slug"`
	Description string    `json:"description" db:"description"`
	IsPersonal  bool      `json:"is_personal" db:"is_personal"`
	Role        string    `json:"role"`         // User's role in this team (not stored in DB, populated at runtime)
	MemberCount int       `json:"member_count"` // Number of members in this team (not stored in DB, populated at runtime)
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

// TeamStatsResponse holds team-wide resource counts for the team analytics page.
// All counts reflect items that belong to the team regardless of their status.
type TeamStatsResponse struct {
	TotalProjects   int `json:"total_projects"`
	TotalPrompts    int `json:"total_prompts"`
	TotalArtifacts  int `json:"total_artifacts"`
	TotalBlueprints int `json:"total_blueprints"`
	TotalMemories   int `json:"total_memories"`
	TotalFeedItems  int `json:"total_feed_items"`
}

// TeamResourceCreationCount is a sparse per-day creation count for a single
// resource type within a team, as returned by the repository before the handler
// zero-fills it into a continuous daily series. ResourceType is one of:
// "prompts", "artifacts", "blueprints", "memories", "projects". Date is a
// YYYY-MM-DD string.
type TeamResourceCreationCount struct {
	Date         string
	ResourceType string
	Count        int
}

// TeamFeedCreationCount is a sparse per-day creation count for a single feed
// entity kind within a team, as returned by the repository before the handler
// zero-fills it into a continuous daily series. EntityType is one of "feeds"
// (channels) or "feed_items" (AI updates posted). Date is a YYYY-MM-DD string.
type TeamFeedCreationCount struct {
	Date       string
	EntityType string
	Count      int
}

// CreateTeamRequest represents the request body for creating a team
type CreateTeamRequest struct {
	Name        string `json:"name"`
	Description string `json:"description"`
}

// UpdateTeamRequest represents the request body for updating a team
type UpdateTeamRequest struct {
	Name        *string `json:"name,omitempty"`
	Description *string `json:"description,omitempty"`
}

// TeamListResponse represents the response for listing teams
type TeamListResponse struct {
	Teams      []Team `json:"teams"`
	TotalCount int    `json:"total_count"`
	Page       int    `json:"page"`
	PageSize   int    `json:"page_size"`
}

// TeamMemberDetail represents detailed information about a team member
// including user information and invitation status
type TeamMemberDetail struct {
	UserID           string  `json:"user_id"`
	Email            string  `json:"email"`
	Name             string  `json:"name"`
	Role             string  `json:"role"`
	JoinedAt         string  `json:"joined_at"`
	InvitationStatus *string `json:"invitation_status,omitempty"` // "pending" or "accepted"
}

// TeamMembersListResponse represents the response for listing team members
type TeamMembersListResponse struct {
	Members    []TeamMemberDetail `json:"members"`
	TotalCount int                `json:"total_count"`
	Page       int                `json:"page"`
	PageSize   int                `json:"page_size"`
}
